File: ClientApp/src/app/services/orders-service/order.ts
import { Customer } from "../customers-service/customer";
import { Pizza } from "../pizza-service/pizza";

export interface Order {
  orderNumber?: string;
  type: string;
  date?: number;
  customer: Customer;
  pizza: Pizza;
  quantity: number;
}

export enum OrderSatus {
  Registered = 1,
  Preparation = 2,
  ReadyToDeliver = 3,
  Delivered = 4
}

File: Data/Entities/Order.cs
using OneTwoThreePizzaApp.Data.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace OneTwoThreePizzaApp.Data.Entities
{
    public class Order
    {
        [Key]
        public Guid OrderNumber { get; set; }     
        public DateTime Date { get; set; }
        public Customer Customer { get; set; }
        public Pizza Pizza { get; set; }
        public OrderSatus OrderStatus { get; set; }
        public int Quantity { get; set; }
    }
}
